component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java

```diff
@@ -1,62 +1,95 @@
+    super(wikiRef);
+  }
+
+  public CalendarEventSearchQuery(IEventSearchQuery query) {
+    super(query, null);
+  }
 
-  private final Date startDate;
-  private final boolean isArchive;
-  private final List<String> allowedSpaces;
-  
+  @Deprecated
   public CalendarEventSearchQuery(ICalendar cal, List<String> sortFields) {
     super(cal.getWikiRef(), getDefaultSortFields(sortFields, cal.isArchive()));
     this.startDate = cal.getStartDate();
     this.isArchive = cal.isArchive();
     this.allowedSpaces = cal.getAllowedSpaces();
   }
 
+  @Deprecated
   public CalendarEventSearchQuery(ICalendar cal, IEventSearchQuery query) {
     super(query, getDefaultSortFields(null, cal.isArchive()));
     this.startDate = cal.getStartDate();
     this.isArchive = cal.isArchive();
     this.allowedSpaces = cal.getAllowedSpaces();
   }
 
+  @Override
+  public ICalendarSearchQueryBuilder addCalendarRestrictions(ICalendar cal) {
+    checkNotNull(cal);
+    this.startDate = cal.getStartDate();
+    this.isArchive = cal.isArchive();
+    this.allowedSpaces = cal.getAllowedSpaces();
+    setSortFields(getDefaultSortFields(getSortFields(), cal.isArchive()));
+    return this;
+  }
+
   @Override
   protected LuceneQuery getAsLuceneQueryInternal(LuceneQuery query) {
     query = super.getAsLuceneQueryInternal(query);
-    query.add(getSearchService().createRestrictionGroup(Type.OR, Arrays.asList("space"), 
+    return addCalendarRestrictions(query);
+  }
+
+  private LuceneQuery addCalendarRestrictions(LuceneQuery query) {
+    checkState(this.startDate != null,
+        "Calendar must be initialized before calling addCalendarRestrictions");
+    query.add(getSearchService().createRestrictionGroup(Type.OR, Arrays.asList("space"),
         getAllowedSpaces()));
     if (!isArchive) {
       query.add(getSearchService().createFromDateRestriction(
-          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "." 
-          + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, true));
+          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "."
+              + ICalendarClassConfig.PROPERTY_EVENT_DATE,
+          startDate, true));
     } else {
-      query.add(getSearchService().createToDateRestriction(
-          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "." 
-          + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, false));
+      query.add(getSearchService().createToDateRestriction(ICalendarClassConfig.CALENDAR_EVENT_CLASS
+          + "." + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, false));
     }
     return query;
   }
 
-  private List<String> getAllowedSpaces() {
-    List<String> ret = new ArrayList<String>();
-    if (allowedSpaces != null && allowedSpaces.size() > 0) {
-      for(String space : allowedSpaces) {
+  private @NotNull List<String> getAllowedSpaces() {
+    List<String> ret = new ArrayList<>();
+    if (!allowedSpaces.isEmpty()) {
+      for (String space : allowedSpaces) {
```

Comment:
This method can be simplified with centralized logic in LuceneUtils, e.g.:
```
List<String> ret = LuceneUtils.exactify(allowedSpaces);
if (ret.isEmpty()) {
  // inexistent space
  ret.add("\".\"");
}
return ret
```

Reply:
this is not the same. Exactify is handling a string and getAllowedSpaces is handling an array of Strings. I did not touch this method and do not see why I should risk to brake it. (not enought direct unit testing available to guard a refactoring)

Reply:
it is the same since exactify is overloaded for a List<String>. you already touched the method, but we can leave it be if you consider it too dangerous.